File: frontend/src/types/technician.ts
import { Region } from "./region";
import { User } from "./user";

export interface Technician {
  id: number;
  user: User;
  region: Region | null;
  whatsappNumber: string | null;
  currentWorkload: number;
  isAvailable: boolean;
  createdAt: string;
  updatedAt: string;
}

/** POST /api/technicians — User + Technician birlikte oluşturulur */
export interface CreateTechnicianRequest {
  fullName: string;
  email: string;
  password: string;
  phone?: string;
  whatsappNumber?: string;
  regionId?: number | null;
  currentWorkload?: number;
  isAvailable?: boolean;
}

/** PUT /api/technicians/{id} — mevcut entity alanları */
export interface UpdateTechnicianRequest {
  user?: { id: number };
  region?: { id: number } | null;
  whatsappNumber?: string;
  currentWorkload?: number;
  isAvailable?: boolean;
}

File: backend/src/main/java/com/servis/backend/entity/Technician.java
package com.servis.backend.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "technicians")
public class Technician {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @OneToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user; // Teknisyenin kullanıcı hesabı

    @ManyToOne
    @JoinColumn(name = "region_id")
    private Region region; // Teknisyenin çalıştığı bölge

    @Column(name = "whatsapp_number", length = 20)
    private String whatsappNumber;

    @Column(name = "current_workload")
    private Integer currentWorkload = 0; // Mevcut iş yükü (atandığı açık iş emirleri)

    @Column(name = "is_available")
    private Boolean isAvailable = true; // Müsaitlik durumu

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}